### PR TITLE
Bug Fix: Check that a new character has a unique name

### PR DESCRIPTION
**Pull request by dyl10s**
This is my first pull request. I tried to take on something simple. I am very open to feedback on this. My changes fix issue #180 with a generic error if the name already exists

### CODE REVIEW

File: Source/NexusForever.WorldServer/Database/Character/CharacterDatabase.cs [hunk 1]
```diff
@@ -87,6 +87,14 @@ public static async Task<List<Model.Character>> GetCharacters(uint accountId)
             }
         }
 
+        public static bool CharacterNameExists(string characterName)
+        {
+            using (var context = new CharacterContext())
+            {
+                return context.Character.Where(c => c.Name == characterName).Count() == 0 ? false : true;
```
Comment (Fabi) [inline Source/NexusForever.WorldServer/Database/Character/CharacterDatabase.cs]: If you do it like that, you can use  `context.Character.Any(c => c.Name == characterName);`

Comment (dyl10s) [reply Source/NexusForever.WorldServer/Database/Character/CharacterDatabase.cs]: That seems more readable. This is entity framework correct? I have never used it before.

**Comment by kirmmin** [comment]
Good job on the PR, @dyl10s . 

I'd encourage you to use appropriate result codes. They are available here: https://gist.github.com/kirmmin/1b632efc95abb791d0fdc070fe486f20. This file does not exist in the codebase at the moment, it's a dump from the `CodeEnumCharacterModifyResult` enum in the Client's code, so please include this file as part of your PR. I would also suggest updating the packet handler to use this Enum, too.

**Comment by dyl10s** [comment]
> CodeEnumCharacterModifyResult

@kirmmin 
Is this what you meant by updating the packet handler?

**Comment by kirmmin** [comment]
> > CodeEnumCharacterModifyResult
> 
> Is this what you meant by updating the packet handler?

Exactly right. Good job!

**Pull request merged**